src/messaging/utils.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { ICType } from '@kiltprotocol/types'
+import { IMessage, IMessageBodyBase } from '../types'
+import * as MessageError from './Error'
+import { CType } from '@kiltprotocol/sdk-js'
+import { UUID } from '@kiltprotocol/utils'
+
+/**
+ * Constructs a message from a message body.
+ * This should be encrypted with [[encrypt]] before sending to the receiver.
+ *
+ * @param body The body of the message.
+ * @param sender The DID of the sender.
+ * @param receiver The DID of the receiver.
+ * @returns The message created.
+ */
+
+export function fromBody<Body extends IMessageBodyBase>(
+  body: Body,
+  sender: IMessage['sender'],
+  receiver: IMessage['receiver']
+): IMessage {
```

Comment:
Is it so not simpler? 
`export function fromBody(
  body: MessageBody,
  sender: IMessage['sender'],
  receiver: IMessage['receiver']
): IMessage<MessageBody> {
`

Reply:
> Is it so not simpler?
> 
> ```
> export function fromBody(
>   body: MessageBody,
>   sender: IMessage['sender'],
>   receiver: IMessage['receiver']
> ): IMessage<MessageBody> {
> ```

yes but it would constrain message creation to the credential api types again. Using the generics syntax makes it more ... generic. It conserves the type of the exact Message body after creation.

Reply:
fair point: [7d348a0](https://github.com/KILTprotocol/kilt-extension-api/pull/18/commits/7d348a036e45e46627abe0693ed3080b841d5885)